File: BC - w4/Date/date.cpp
#include <iostream>
#include "date.h"



        Date::Date(int day, int month, int year): 
        day(day),
        month(month),
        year(year)
        {
            validate(day, month, year);
        }

        Date::~Date(){}

        int Date::getDay() const{
            return this->day;
        }
        int Date::getMonth() const{
            return this->month;
        }
        int Date::getYear() const{
            return this->year;
        }
        void Date::validate(int day, int month, int year) {
        int chek;
        int LeapYear;

        if ( month > 12 || month < 1) {
            throw InvalidDate ("invalid month");
        }

        LeapYear = this->year % 4;

        chek = (this->month % 2) ^ (this->month / 8);

            if ( chek == 1 ) {
                if (this->day > 31) {
                    throw InvalidDate ("invalid day");
                };
            } else if ( this->month == 2 ) {
                if ( LeapYear == 0) {
                    if (this->day > 28) {
                        throw InvalidDate("invalid day");
                    }
                } else {
                    if (this->day > 29) {
                        throw InvalidDate("invalid day");
                    }
                }
        } else {
                if (this->day > 30) {
                    throw InvalidDate("invalid day");
                }
            }
    }

std::ostream& operator<<(std::ostream& out, const Date& date){
    if ( date.getDay() < 10) {
        out<<"Day 0"<< date.getDay() <<std::endl;
    } else {
        out<<"Day "<< date.getDay() <<std::endl;
    }
    out<<"Month "<< date.getMonth() <<std::endl;
    out<<"Year "<< date.getYear() <<std::endl;
}

File: BC - w4/java/Date/Date.java
package Date;

public class Date {
	private int day;
	private int month;
	private int year;
	
	public Date(int day, int month, int year) throws InvalidDate {
		this.day = day;
		this.month = month;
		this.year = year;
        validate(day, month, year);
        }
	
	
	public int getDay() {
		return day;
	}

	public int getMonth() {
		return month;
	}
	
	public int getYear() {
		return year;
	}
	
	public void validate(int day, int month, int year) throws InvalidDate {
    int chek;
    int LeapYear;

    if ( month > 12 || month < 1) {
        throw new InvalidDate ("invalid month");
    }

    LeapYear = this.year % 4;

    chek = (this.month % 2) ^ (this.month / 8);

        if ( chek == 1 ) {
            if (this.day > 31) {
                throw new InvalidDate ("invalid day");
            };
        } else if ( this.month == 2 ) {
            if ( LeapYear == 0) {
                if (this.day > 28) {
                    throw new InvalidDate("invalid day");
                }
            } else {
                if (this.day > 29) {
                    throw new InvalidDate("invalid day");
                }
            }
    } else {
            if (this.day > 30) {
                throw new InvalidDate("invalid day");
            }
        }
}
	
	
	public void print(){
		System.out.println("day "+this.day+"\n"+ "month " + this.month+"\n"+"year " + this.year);
	}

}
